lib/elastic_apm/spies/sneakers.rb

```diff
@@ -0,0 +1,31 @@
+module ElasticAPM
+  # @api private
+  module Spies
+    # @api private
+    class SneakersSpy
+      def install
+        Sneakers.middleware.use(Middleware, nil)
+      end
+      #
+      class Middleware
+        def initialize(app, *args)
+          @app = app
+          @args = args
+        end
+
+        def call(deserialized_msg, delivery_info, metadata, handler)
+          transaction = ElasticAPM.start_transaction(delivery_info.routing_key , 'Sneakers')
```

Comment:
Do we want the first argument to `start_transaction` (i.e. the name) to be something related to the message? Then have the label on the transaction be the `routing_key`. 
I'm asking because in the Sidekiq spy, we use the job's class name as the name for the transaction and set a label for the queue identifier. 
Perhaps you can explain what the `routing_key`'s significance is and we can see what attribute on the transaction suits it best. For example, if it's something that changes a lot, perhaps it'd be better to have it be a label.

Reply:
you are absolutely right. The problem is that in Sneakers, the middleware doesn't have access to the worker class/name. I opened a pull request to change that https://github.com/jondot/sneakers/pull/414

That is why I used routing_key as a replacement for now.

Reply:
@estolfo  I changed the sneakers spy to use queue name for transaction and uses the routing_key as a label